### PR TITLE
Removed the XR pattern and added the initiator check in WriteMessage and ReadMessage according to spec

### PR DESCRIPTION
**Pull request by mimoo**
Tests passed, I tried interacting with my own implementation and it works as well ✅

### CODE REVIEW

File: vectors.txt [hunk 1]
```diff
@@ -19756,12599 +17902,3 @@ msg_1_ciphertext=131094feaa8cbacb6c348050711162cf8d44b13e8de882c98410e7d3981d51
...
-
-handshake=Noise_NK_25519_ChaChaPoly_SHA512
-resp_static=0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20
-gen_init_ephemeral=202122232425262728292a2b2c2d2e2f303132333435363738393a3b3c3d3e3f
-gen_resp_ephemeral=4142434445464748494a4b4c4d4e4f505152535455565758595a5b5c5d5e5f60
-msg_0_payload=746573745f6d73675f30
-msg_0_ciphertext=358072d6365880d1aeea329adf9121383851ed21a28e3b75e965d0d2cd16625490f8f004794122bca779d9bedcc39b7bfb88883f6a56029320aa
-msg_1_payload=746573745f6d73675f31
-msg_1_ciphertext=64b101b1d0be5a8704bd078f9895001fc03e8e9f9522f188dd128d9846d48466e93b0314d9d7741d4e27466c0dd7656cf06185f352ffe422a67d
-msg_2_payload=79656c6c6f777375626d61726
```
Comment (titanous) [inline vectors.txt]: Any idea what happened to all of these test vectors?

Comment (mimoo) [reply vectors.txt]: I removed all the text vectors associated to the `XR` pattern :o did I mess up with something?

Comment (titanous) [reply vectors.txt]: This one looks like it was a NK pattern?

Comment (mimoo) [reply vectors.txt]: gosh yeah, looks like my regex was flawed. Let me try again

**Comment by mimoo** [comment]
Alright, I've pushed the fix. Maybe a better way to verify this would be to have you do the same and diff my and your `vectors.txt`?

**Comment by mimoo** [comment]
Oh wait, I just realized the `vectorgen.go` is the file that generates the vectors.txt -_-

**Comment by mimoo** [comment]
Alright, re-generated the file via `vectorgen.go` and it confirmed that I did not made any error the second time. It also removed the first line of comment:

```py
# Vectors generated by https://github.com/titanous/noise
```

**Comment by titanous** [comment]
Thanks!
